src/test/java/dog/pawbook/logic/commands/ScheduleCommandTest.java

```diff
@@ -0,0 +1,82 @@
+package dog.pawbook.logic.commands;
+
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_NAME_OBEDIENCE_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_SESSION_OBEDIENCE_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_TAG_ALL;
+import static dog.pawbook.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static dog.pawbook.model.managedentity.IsEntityPredicate.IS_PROGRAM_PREDICATE;
+import static dog.pawbook.testutil.Assert.assertThrows;
+import static dog.pawbook.testutil.TypicalEntities.getTypicalDatabase;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.model.Model;
+import dog.pawbook.model.ModelManager;
+import dog.pawbook.model.UserPrefs;
+import dog.pawbook.model.managedentity.ProgramOccursOnDatePredicate;
+import dog.pawbook.model.managedentity.program.Program;
+import dog.pawbook.testutil.ProgramBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for ListCommand.
+ */
+public class ScheduleCommandTest {
+
+    private Model model = new ModelManager(getTypicalDatabase(), new UserPrefs());
+    private Model expectedModel = new ModelManager(model.getDatabase(), new UserPrefs());
+
+    @BeforeEach
+    public void setUp() {
+        model.updateFilteredEntityList(x -> x.getValue() instanceof Program);
+    }
+
+    @Test
+    public void execute_emptyArguments_success() {
+
+        // Credit: https://www.baeldung.com/java-override-system-time
+        Clock clock = Clock.fixed(Instant.parse("2022-01-01T00:00:00Z"), ZoneId.of("UTC"));
+        LocalDate testDate = LocalDate.now(clock);
```

Comment:
I think the issue with this is that our internal functional code does not make use of the same clock, it is either that we modify our code to support the use of a specified clock or we just add a program with a session date and time that happens on the same day instead, then we can use it to verify. I don't think these 2 variables are being used.

Reply:
Fixed.